src/data_chain/mod.rs

```diff
@@ -20,9 +20,92 @@ mod chain;
 mod node_state;
 mod proof;
 mod vote;
+mod vote_accumulator;
 
-pub use self::block::{Block, NodesAndAge};
+pub use self::block::{Block, BlockState, NodesAndAge};
 pub use self::chain::Chain;
 pub use self::node_state::{NodeState, State};
 pub use self::proof::Proof;
 pub use self::vote::Vote;
+pub use self::vote_accumulator::VoteAccumulator;
+
+use public_info::PublicInfo;
+use std::collections::BTreeSet;
+
+// FIXME - remove
+#[allow(unused)]
+/// Calculates whether a quorum of nodes have voted.  In this case, "quorum" means >50% of the
+/// members in `valid_nodes_itr` are included in `voters_itr` and that their cumulative age is >50%
+/// of the cumulative age of all members of `valid_nodes_itr`.
+pub fn quorum<'a, 'b, I, J>(voters_itr: I, valid_nodes_itr: J) -> bool
```

Comment:
Seems we can use this function in `Block::get_block_state`. Possibly other places too. Not sure it should be done in this PR or other, so feel free to dismiss.